### PR TITLE
Add salesforce configuration difftool

### PR DESCRIPTION
**Pull request by DanielHeath**

### CODE REVIEW

File: lib/connect/api_adapter.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module Connect
+  class ApiAdapter
+    API_VERSION = "41.0".freeze
```
Comment (matenia) [inline lib/connect/api_adapter.rb]: Do we perhaps need to ENV this? maybe with a default?

File: lib/tasks/diff_salesforce.rake [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        diff = true
+      end
+
+      HashDiff.diff(old_h.except(*removed), new_h.except(*added)).each do |sym, key, old, new_val|
+        diff = true
+        case sym
+        when "~"
+          puts "Changed #{key} from #{old} to #{new_val}"
+        when "-"
+          puts "Removed #{key}"
```
Comment (MC-Squared) [inline lib/tasks/diff_salesforce.rake:51]: I'd have thought `-` and `+` are standard enough diff prefixes to just use them as-is?
Also, not sure if the output would be nicer if it were sorted first? (i.e. all removals, then all additions, etc). I guess changes are generally small so probably doesn't matter.

**Review by MC-Squared — APPROVED** [review]

File: lib/tasks/diff_salesforce.rake (hunk 2 above)
Comment (DanielHeath) [reply lib/tasks/diff_salesforce.rake:51]: RE `+/-/~`: It's nice to be able to copy-paste to a salesforce specialist on the other side, who might not be familiar with `diff`.

I've sorted table addition/removal but not fields (yet) - but yes, changes are generally pretty small.

File: lib/connect/api_adapter.rb (hunk 1 above)
Comment (DanielHeath) [reply lib/connect/api_adapter.rb]: Yeah, that makes sense.

**Pull request merged**